torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -3533,10 +3573,24 @@ def forward(self, x, rank):
                         model=test_input.model,
                         inp=test_input.inp,
                         sync_interval=i + 2,
+                        throw_on_early_termination=test_input.throw_on_early_termination,
+                    )
+                )
+
+            throw_on_early_term_tests = []
+            for test_input in models_to_test:
```

Comment:
Might be nice to add a test with SyncBatchNorm to verify this new option works for such models too.